### DISCUSSION TITLE
ESP32-CAM image quality

### DISCUSSION THREAD

**Discussion by chobott** [jomjol/AI-on-the-edge-device#1057] [General] [answered]
Hi everybody !

i have install ESP32-CAM with 2Mpx Camera but picture is everytime horibble :-) in configuration i cannot change better then VGA and probably Camera can make autofocus? Image attached

What is best way how i can make better image

![alg_roi](https://user-images.githubusercontent.com/24911627/191939876-2c48cd36-2fca-4eb9-8db0-c82b0c0a0423.jpg)

**Comment by friedpa** [comment, marked as answer]
Please read Wiki how to setup the camera.

**Reply by chobott** [reply]
i can try lot of setting, i have printed original tube....but if check project page there is beatifull pictures 


![Screenshot 2022-09-23 at 12-54-34 Home · jomjol_AI-on-the-edge-device Wiki](https://user-images.githubusercontent.com/24911627/191946148-c10ee341-4251-4c3b-b106-91a4fa0e65e0.png)

**Comment by chobott** [comment]
This is screen from OV2640 setting from another project
![ZGJiTaa](https://user-images.githubusercontent.com/24911627/191947260-c2291963-53f8-4037-9996-b90b7b428169.png)

**Reply by friedpa** [reply]
https://github.com/jomjol/AI-on-the-edge-device/wiki/Installation
![grafik](https://user-images.githubusercontent.com/42890695/191950403-b1c3f43b-f912-4a05-8490-e59cf71fcbfe.png)

**Reply by chobott** [reply]
OK after change camera to another all working fine :-) 

I may have found a bug when setting up the new camera, if I disable field MQTT "URL" in the configuration, then ESP restarts all the time, i must get config file into SD card and change back

![alg_roi](https://user-images.githubusercontent.com/24911627/192050497-1bb8ad9c-de50-4262-b85a-44e9b743d20c.jpg)
